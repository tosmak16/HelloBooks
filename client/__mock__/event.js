export const e = {
  target: {
    name: 'Tosmak',
    value: '10'
  },
};

export class event {
  static preventDefault() {

  }
}

