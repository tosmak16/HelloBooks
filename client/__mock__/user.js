const user = [{
  firstName: 'Tobi',
  lastName: 'Akinola',
  email: 'tobi@gmail.com',
  mobileNumber: '0803723455',
  membershipType: 'Basic',
  image: 'user.jpg',
  profileImage: 'user.jpg'
}];

export default user;
