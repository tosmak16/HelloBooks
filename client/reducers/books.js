import { GET_ALL_BOOKS_FAILURE, GET_ALL_BOOKS_SUCCESS, GET_ALL_BOOKS_REQUEST } from '../actions/getBooks';

const initialState = {
  isFetching: false,
  isFetched: false,
  data: [],
  error: '',
};
export default (state = initialState, action = {}) => {
  switch (action.type) {
    case GET_ALL_BOOKS_REQUEST:
      return {
        isFetching: true,
        isFetched: false,
        data: [],
        error: '',
      };
    case GET_ALL_BOOKS_SUCCESS:
      return {
        isFetching: false,
        isFetched: true,
        data: action.data,
        error: '',
      };
    case GET_ALL_BOOKS_FAILURE:
      return {
        isFetching: false,
        isFetched: false,
        data: [],
        error: action.error,
      };
    default:
      return state;
  }
};
