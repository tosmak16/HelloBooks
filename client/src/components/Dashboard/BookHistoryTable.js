import React from 'react';
import isEmpty from 'lodash/isEmpty';
import filterBy from 'lodash/filter';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import BhTableRow from '../Dashboard/BhTableRow';
import getborrowedBooks from '../../actions/getborrowedBooks';
import getbooks from '../../actions/getBooks';
import refreshPage from '../../actions/refreshPage';

let tablerow = '';
class BorrowHistoryTable extends React.Component {
  componentWillMount() {
    if (isEmpty(this.props.data)) {
      this.props.getborrowedBooks();
    }

    if (isEmpty(this.props.bookData)) {
      this.props.getbooks(true);
    }
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.isRefreshed) {
      setTimeout(() => { this.props.refreshPage(true); }, 1000);
      this.props.getborrowedBooks();
      setTimeout(() => { this.props.refreshPage(false); }, 1000);
      // this.props.getbooks(true);
    }
  }
  render() {
    if (this.props.data) {
      const { data } = this.props;
      tablerow = data.map(row =>
        (<BhTableRow
          key={ row.id }
          row={ row }
          value={ row.id }
          bookItem={ filterBy(this.props.bookData, ['id', row.bookId]) }
        />)
      );
    }
    return (
      <div id="bh_table" className="row">
        <div className=" col l10 offset-l2 col m10 offset-m2 col s12">
          <h4 className="sub-header">Read List</h4>
          <div className="table-responsive">
            <table id="table_bh" className="table responsive-table bordered highlight striped">
              <thead>
                <tr>
                  <th><span className="glyphicon glyphicon-education" /></th>
                  <th>Title</th>
                  <th>Author</th>
                  <th>Borrowed Date</th>
                  <th>Returned Date</th>
                </tr>
              </thead>
              <tbody>
                {tablerow}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    );
  }
}

BorrowHistoryTable.propTypes = {
  getbooks: PropTypes.func.isRequired,
  getborrowedBooks: PropTypes.func.isRequired,
  refreshPage: PropTypes.func.isRequired,

};

function mapStateToProps(state) {
  return {
    error: state.getborrowedBooks[0].error,
    data: state.getborrowedBooks[0].data,
    bookData: state.books[0].data,
    isRefreshed: state.refreshPage[0].isRefreshed
  };
}

export default connect(mapStateToProps, { getborrowedBooks, getbooks, refreshPage })(BorrowHistoryTable);
