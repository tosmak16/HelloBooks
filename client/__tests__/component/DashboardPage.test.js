import React from 'react';
import { mount, shallow } from 'enzyme';
import render from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import expect from 'expect';
import thunk from 'redux-thunk';
import jwt from 'jsonwebtoken';


import localStorageMock from '../../__mock__/localStorage';
import books from '../../__mock__/book';
import user from '../../__mock__/user';
import { e, event } from '../../__mock__/event';
import { UserProfileForm } from '../../src/components/Dashboard/UserProfileForm';
import { BorrowHistoryTable } from '../../src/components/Dashboard/BookHistoryTable';
import { BorrowedbooksTable } from '../../src/components/Dashboard/BorrowedbooksTable';
import { ChangePasswordForm } from '../../src/components/Dashboard/ChangePasswordForm';
import { DashboardSidebar } from '../../src/components/Dashboard/DashboardSidebar';
import BorrowHistoryBooksPageContainer, { BorrowHistoryBooksPage } from '../../src/components/Dashboard/Containers/BorrowHistoryBooksPage';
import ChangePasswordPageContainer, { ChangePasswordPage } from '../../src/components/Dashboard/Containers/ChangePasswordPage';
import UserProfilePageContainer, { UserProfile } from '../../src/components/Dashboard/Containers/UserProfile';
import BorrowedBooksPageContainer, { BorrowedBooksPage } from '../../src/components/Dashboard/Containers/BorrowedBooksPage';
import { BookHistoryTableRow } from '../../src/components/Dashboard/BookHistoryTableRow';

jest.mock('jsdom');
jest.mock('jwt-decode');
jest.mock('../../__mock__/localStorage');
const middlewares = [thunk]; // add your middlewares like `redux-thunk

window.document = localStorageMock;

const mockFuction = jest.fn();
const token = jwt.sign({ id: 1, user: 'Tosmak', role: 'admin' }, 'encoded');
window.localStorage = localStorageMock;
localStorage.setItem('jwtToken', token);


describe('Test Dashboard page components and container', () => {
  const initialState = {
    books: [{
      isFetching: false,
      isFetched: false,
      data: books,
      error: '',
    }],
    UserDetails: [{
      isFetching: false,
      data: [user[0]],
      error: '',
    }],
    refreshPage: [{
      isRefreshed: false,
    }],
    getborrowedBooks: [{
      isFetching: false,
      isFetched: false,
      data: books,
      error: '',
    }],
    returnBooks: [{
      data: '',
      error: '',
      isReturning: false,
      response: '',
    }],
    getunreturnedBooks: [{
      isFetching: false,
      isFetched: false,
      data: books,
      error: '',
    }],
    updateUser: [{
      isUpdating: false,
      data: '',
      error: '',
    }],
    userProfileImage: [{
      data: '',
      error: '',
      isUploading: false,
      response: '',
    }],
    passwordChange: [{
      isSending: false,
      data: '',
      error: '',
    }],
  };
  const mockStore = configureMockStore(middlewares);
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    window.localStorage = localStorageMock;
    //  wrapper = mount(<Provider store={store}><BorrowedBooksPage /></Provider>);
  });


  it('should test and take snapshot of BorrowedbooksTable', () => {
    const tree = render.create(<BorrowedbooksTable
      bookData={[]}
      unreturnedBooksData={[]}
      getunreturnedBooks={mockFuction}
      getbooks={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of BorrowHistoryTable', () => {
    const tree = render.create(<BorrowHistoryTable
      bookData={[]}
      unreturnedBooksData={[]}
      getborrowedBooks={mockFuction}
      getbooks={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of ChangePasswordForm', () => {
    const tree = render.create(<ChangePasswordForm
      passwordChange={[]}
      changePassword={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of DashboardSidebar', () => {
    const tree = render.create(<DashboardSidebar
      userData={[]}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should test and take snapshot of UserProfileForm', () => {
    const tree = render.create(<UserProfileForm
      userData={[]}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should test and take snapshot of BorrowedBooksPage', () => {
    const tree = render.create(<BorrowedBooksPage
      getUserdetails={mockFuction}
      getunreturnedBooks={mockFuction}
      getbooks={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should test and take snapshot of BorrowHistoryBooksPage', () => {
    const tree = render.create(<BorrowHistoryBooksPage
      getborrowedBooks={mockFuction}
      getUserdetails={mockFuction}
      getbooks={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should test and take snapshot of ChangePasswordPage', () => {
    const tree = render.create(<ChangePasswordPage
      getborrowedBooks={mockFuction}
      getUserdetails={mockFuction}
      getbooks={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should test and take snapshot of UserProfile', () => {
    const tree = render.create(<UserProfile
      getborrowedBooks={mockFuction}
      getUserdetails={mockFuction}
      getbooks={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });
})
