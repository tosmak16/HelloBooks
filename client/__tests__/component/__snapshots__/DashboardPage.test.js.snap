// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard page components and container should test and take snapshot of dashboard Page 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s12 col m12 col l12"
    >
      <div
        className="row"
      >
        <div
          className="col s12 col m2 col l2"
        >
          <div
            id="sidebar-wrapper"
          >
            <ul
              className="sidebar-nav"
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <li>
                <div
                  id="imageborder"
                >
                  <div
                    className="file-field input-field"
                  >
                    <div
                      className=""
                      id="filebtn"
                      style={
                        Object {
                          "color": "white",
                          "marginLeft": "5px",
                        }
                      }
                    >
                      <span />
                      <img
                        alt="images"
                        height="120"
                        id="userimg"
                        src="http://res.cloudinary.com/tosmak/image/upload/v1507297483/userimg_cxeszl.png"
                        width="120"
                      />
                      <input
                        accept=".png, .jpg, .jpeg"
                        className="fileInput"
                        disabled={undefined}
                        id="photoInput"
                        onChange={[Function]}
                        placeholder="Upload profile image"
                        type="file"
                      />
                    </div>
                    <div
                      className="file-path-wrapper"
                    >
                      <input
                        className="file-path validate"
                        placeholder="Upload profile image"
                        style={
                          Object {
                            "width": "100px",
                          }
                        }
                        type="text"
                      />
                      <div
                        className="input-field inline"
                      >
                        <button
                          className="btn-sm"
                          id=""
                          onClick={[Function]}
                          style={
                            Object {
                              "backgroundColor": "#3d1112",
                              "color": "wheat",
                              "marginTop": "0px",
                            }
                          }
                          type="button"
                        >
                          Upload
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <a
                  href="#"
                  id="dash"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons left"
                  >
                    local_library
                  </i>
                   Books Shelf
                </a>
              </li>
              <li>
                <a
                  href="#"
                  id="myP"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons left"
                  >
                    dashboard
                  </i>
                  Account
                </a>
              </li>
              <li>
                <a
                  href="#"
                  id="bor"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons left"
                  >
                    data_usage
                  </i>
                  History
                </a>
              </li>
              <li>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons left"
                  >
                    lock
                  </i>
                  Change Password
                </a>
              </li>
              <li>
                <a
                  href="index.html"
                >
                  <i
                    className="material-icons left"
                  >
                    exit_to_app
                  </i>
                  Logout
                </a>
              </li>
            </ul>
          </div>
          <div
            className="modal"
            id="modaE"
          >
            <div
              className="modal-content"
            >
              <h5>
                Oh!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modaS"
          >
            <div
              className="modal-content"
            >
              <h5>
                Done!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modaO"
          >
            <div
              className="modal-content"
            >
              <h5>
                Do you want to change your profile picture?
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                NO
              </a>
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
                type="submit"
              >
                YES
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
        id="bb_table"
      >
        <div
          className="  col l10 offset-l2 col m10 offset-m2 col s12"
        >
          <h4
            className="sub-header"
          >
             Currently Reads
          </h4>
          <div
            className="responsive-table"
          >
            <table
              className="table responsive-table bordered highlight striped"
            >
              <thead>
                <tr>
                  <th>
                    <span
                      className="glyphicon glyphicon-education"
                    />
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    ISBN
                  </th>
                  <th>
                    Action
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE PEOPLE IN THE TREE
                  </td>
                  <td>
                    Y. Hayan
                  </td>
                  <td>
                    Adventure
                  </td>
                  <td>
                    345798465
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="1,1"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE LATE PARADE
                  </td>
                  <td>
                    F. Adam
                  </td>
                  <td>
                    Adventure
                  </td>
                  <td>
                    234560878
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="2,2"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    I AM THE WOLF
                  </td>
                  <td>
                    L. Mark
                  </td>
                  <td>
                    Music
                  </td>
                  <td>
                    4678980976
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="3,3"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE PALE NORTH
                  </td>
                  <td>
                    H. Clayton
                  </td>
                  <td>
                    Adventure
                  </td>
                  <td>
                    34567887655
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="4,4"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    ADULTOLESENCE
                  </td>
                  <td>
                    G. Hannah
                  </td>
                  <td>
                    Romance
                  </td>
                  <td>
                    45678974567
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="5,5"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    WHAT IT MEANS WHEN A MAN FALL FROM SKY
                  </td>
                  <td>
                    L. Nneka
                  </td>
                  <td>
                    Adventure
                  </td>
                  <td>
                    567244567
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="6,6"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE SONG AND THE SILENCE
                  </td>
                  <td>
                    Y. Johnson
                  </td>
                  <td>
                    Music
                  </td>
                  <td>
                    64578974567
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="7,7"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE ILIAD HOMMER
                  </td>
                  <td>
                    P. Green
                  </td>
                  <td>
                    Action
                  </td>
                  <td>
                    787890974567
                  </td>
                  <td>
                    <button
                      className="btn-sm btn-success btn-sm returnbtn"
                      name="8,8"
                      onClick={[Function]}
                      type="button"
                    >
                      Return
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            className="modal"
            id="modal3"
          >
            <div
              className="modal-content"
            >
              <h5>
                Done!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modal2"
          >
            <div
              className="modal-content"
            >
              <h5>
                Oh!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modal1"
          >
            <div
              className="modal-content"
            >
              <h5>
                Do you want to return this book?
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                NO
              </a>
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
                type="submit"
              >
                YES
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
        id="b_page"
      >
        <form
          action=""
          className="form-signin col col l10 offset-l1 col m11 offset-m2 col s12"
        >
          <h4
            className="sub-header form-signin-heading"
          >
             Personal Info
          </h4>
          <div
            className="input-field"
          >
            <label
              className="sr-only"
              htmlFor="firstname"
            >
              First Name
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={true}
              id="firstname"
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="input-field"
          >
            <label
              className="sr-only"
              htmlFor="lastname"
            >
              Last Name
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={true}
              id="lastname"
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="input-field"
          >
            <label
              className="sr-only"
              htmlFor="inputSignUpEmail"
            >
              Email address
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={true}
              id="inputSignUpEmail"
              name="email"
              onChange={[Function]}
              placeholder="Email address"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="sr-only"
              htmlFor="inputPhoneNumber"
            >
              Phone number
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={true}
              id="inputPhoneNumber"
              name="mobileNumber"
              onChange={[Function]}
              placeholder="phone number"
              required={true}
              type="number"
              value=""
            />
          </div>
          <select
            className="browser-default"
            disabled={true}
            name="membershipType"
            onChange={[Function]}
          >
            <option
              defaultValue="Basic"
            >
              Basic
            </option>
            <option
              value="Basic"
            >
              Basic
            </option>
            <option
              value="Silver"
            >
              Silver
            </option>
            <option
              value="Gold"
            >
              Gold
            </option>
          </select>
          <div
            className="modal"
            id="modalE"
          >
            <div
              className="modal-content"
            >
              <h5>
                Oh!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modalS"
          >
            <div
              className="modal-content"
            >
              <h5>
                Done!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modalO"
          >
            <div
              className="modal-content"
            >
              <h5>
                Do you want to update your details?
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                NO
              </a>
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
                type="submit"
              >
                YES
              </a>
            </div>
          </div>
          <div
            className="input-field inline"
          >
            <button
              className="btn btn-primary pbtn"
              id="editbtn"
              onClick={[Function]}
              type="button"
            >
              Edit
            </button>
          </div>
        </form>
      </div>
      <div
        className="row"
        id="bh_table"
      >
        <div
          className=" col l10 offset-l2 col m10 offset-m2 col s12"
        >
          <h4
            className="sub-header"
          >
            Read List
          </h4>
          <div
            className="table-responsive"
          >
            <table
              className="table responsive-table bordered highlight striped"
              id="table_bh"
            >
              <thead>
                <tr>
                  <th>
                    <span
                      className="glyphicon glyphicon-education"
                    />
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Borrowed Date
                  </th>
                  <th>
                    Returned Date
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE PEOPLE IN THE TREE
                  </td>
                  <td>
                    Y. Hayan
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE LATE PARADE
                  </td>
                  <td>
                    F. Adam
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    I AM THE WOLF
                  </td>
                  <td>
                    L. Mark
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE PALE NORTH
                  </td>
                  <td>
                    H. Clayton
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    ADULTOLESENCE
                  </td>
                  <td>
                    G. Hannah
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    WHAT IT MEANS WHEN A MAN FALL FROM SKY
                  </td>
                  <td>
                    L. Nneka
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE SONG AND THE SILENCE
                  </td>
                  <td>
                    Y. Johnson
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    Not yet returned
                  </td>
                </tr>
                <tr>
                  <td>
                    .
                  </td>
                  <td>
                     
                    THE ILIAD HOMMER
                  </td>
                  <td>
                    P. Green
                  </td>
                  <td>
                    2017-10-04
                  </td>
                  <td>
                    2017-10-04
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        id="ch_pas"
      >
        <form
          action=""
          className="form-signin col col l10 offset-l1 col m11 offset-m2 col s12"
        >
          <h4
            className="sub-header form-signin-heading"
          >
            Change password
          </h4>
          
          <div
            className="input-field"
          >
            <label
              className="sr-only"
              htmlFor="oldPassword"
            >
              Current Password
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={undefined}
              id="oldPassword"
              name="oldPassword"
              onChange={[Function]}
              placeholder="Current password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="input-field"
          >
            <label
              className="sr-only"
              htmlFor="newPassword"
            >
              New password
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={undefined}
              id="newPassword"
              name="newPassword"
              onChange={[Function]}
              placeholder="New password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="input-field"
          >
            <label
              className="sr-only"
              htmlFor="confirmPassword"
            >
              Confirm password
            </label>
            <input
              autoFocus={true}
              className="form-control validate"
              disabled={undefined}
              id="confirmPassword"
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="modal"
            id="modalError"
          >
            <div
              className="modal-content"
            >
              <h5>
                Oh!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modalSuccess"
          >
            <div
              className="modal-content"
            >
              <h5>
                Done!
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modalOpen"
          >
            <div
              className="modal-content"
            >
              <h5>
                Do you want to change your password?
              </h5>
              <p>
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                NO
              </a>
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
                type="submit"
              >
                YES
              </a>
            </div>
          </div>
          <div
            className="input-field inline"
          >
            <button
              className="btn btn-primary pbtn"
              id="editbtn"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Dashboard page components and container should test and take snapshot of user profile page when data and book array are empty 1`] = `
<div
  className="row"
  id="b_page"
>
  <form
    action=""
    className="form-signin col col l10 offset-l1 col m11 offset-m2 col s12"
  >
    <h4
      className="sub-header form-signin-heading"
    >
       Personal Info
    </h4>
    <div
      className="input-field"
    >
      <label
        className="sr-only"
        htmlFor="firstname"
      >
        First Name
      </label>
      <input
        className="form-control validate"
        disabled={true}
        id="firstname"
        name="firstName"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value="Tobi"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        className="sr-only"
        htmlFor="lastname"
      >
        Last Name
      </label>
      <input
        className="form-control validate"
        disabled={true}
        id="lastname"
        name="lastName"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value="Akinola"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        className="sr-only"
        htmlFor="inputSignUpEmail"
      >
        Email address
      </label>
      <input
        className="form-control validate"
        disabled={true}
        id="inputSignUpEmail"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        required={true}
        type="email"
        value="tobi@gmail.com"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="sr-only"
        htmlFor="inputPhoneNumber"
      >
        Phone number
      </label>
      <input
        className="form-control validate"
        disabled={true}
        id="inputPhoneNumber"
        name="mobileNumber"
        onChange={[Function]}
        placeholder="phone number"
        required={true}
        type="number"
        value="0803723455"
      />
    </div>
    <select
      className="browser-default"
      disabled={true}
      name="membershipType"
      onChange={[Function]}
    >
      <option
        defaultValue="Basic"
      >
        Basic
      </option>
      <option
        value="Basic"
      >
        Basic
      </option>
      <option
        value="Silver"
      >
        Silver
      </option>
      <option
        value="Gold"
      >
        Gold
      </option>
    </select>
    <div
      className="modal"
      id="modalE"
    >
      <div
        className="modal-content"
      >
        <h5>
          Oh!
        </h5>
        <p>
          
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <a
          className="modal-action modal-close waves-effect waves-brown btn-flat"
          href=""
          onClick={[Function]}
        >
          Close
        </a>
      </div>
    </div>
    <div
      className="modal"
      id="modalS"
    >
      <div
        className="modal-content"
      >
        <h5>
          Done!
        </h5>
        <p>
          
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <a
          className="modal-action modal-close waves-effect waves-brown btn-flat"
          href=""
          onClick={[Function]}
        >
          Close
        </a>
      </div>
    </div>
    <div
      className="modal"
      id="modalO"
    >
      <div
        className="modal-content"
      >
        <h5>
          Do you want to update your details?
        </h5>
        <p>
          
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <a
          className="modal-action modal-close waves-effect waves-brown btn-flat"
          href=""
          onClick={[Function]}
        >
          NO
        </a>
        <a
          className="modal-action modal-close waves-effect waves-brown btn-flat"
          href=""
          onClick={[Function]}
          type="submit"
        >
          YES
        </a>
      </div>
    </div>
    <div
      className="input-field inline"
    >
      <button
        className="btn btn-primary pbtn"
        id="editbtn"
        onClick={[Function]}
        type="button"
      >
        Edit
      </button>
    </div>
    <div
      className="activity"
      id="activity-loader-id"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="preloader-wrapper activity-loader big active"
      >
        <div
          className="gap-patch activity-gap2"
        >
          <div
            className="spinner-layer activity-spinner spinner-red-only"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch activity-gap"
            />
            <div
              className="circle"
            />
          </div>
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
